lib/gc.cpp

```diff
@@ -301,7 +301,12 @@ static void gc_callback() {
     }
 }
 
-void silent(char *, GC_word) {}
+// Starting with minor version 3 'GC_set_warn_proc' requires a const pointer.
+#if GC_VERSION_MINOR >= 3
```

Comment:
Maybe this should also check the major version to avoid problems with older and future gc versions? Thinking about something like

```
(GC_VERSION_MAJOR == 8 && GC_VERSION_MINOR >= 3) || GC_VERSION_MAJOR > 8
```